### PR TITLE
Add support for the reject_tokens_expiring_in_less_than option.

### PR DESCRIPTION
**Pull request by ploftness-tesla**
This is a followup pull request, implementing the [simplified approach](https://github.com/Keats/jsonwebtoken/issues/365#issuecomment-1910186315) discussed on  https://github.com/Keats/jsonwebtoken/issues/365.

### CODE REVIEW

**Review by Keats — APPROVED** [review]

File: src/validation.rs [hunk 1]
```diff
@@ -39,6 +39,11 @@ pub struct Validation {
     ///
     /// Defaults to `60`.
     pub leeway: u64,
+    /// Reject a token some time (in seconds) before the `exp` to prevent
```
[the remark this answers is not in the document]
Comment (ploftness-tesla) [reply src/validation.rs:42]: Yes, I have added some explanation.

**Comment by ploftness-tesla** [comment]
@Keats Thanks for approving; when do you think this will be merged/released?

**Comment by ploftness-tesla** [comment]
@Keats I do not have permissions to merge.  What are the next steps for this?

**Comment by Keats** [comment]
I'll merge and release in a bit when i have some free time

**Comment by Keats** [comment]
Looks like there's another MSRV >_>

**Comment by ploftness-tesla** [comment]
> Looks like there's another MSRV >_>

@Keats Yes, looks like the build error is due to a recent change in a transitive dependency used by some tests.  

```
% cargo tree
jsonwebtoken v9.2.0 (/Users/ploftness/Documents/jsonwebtoken)
...
[dev-dependencies]
...
├── time v0.3.31 (*)
└── wasm-bindgen-test v0.3.40
    ├── console_error_panic_hook v0.1.7
    │   ├── cfg-if v1.0.0
    │   └── wasm-bindgen v0.2.90
    │       ├── cfg-if v1.0.0
    │       └── wasm-bindgen-macro v0.2.90 (proc-macro)
    │           ├── quote v1.0.35 (*)
    │           └── wasm-bindgen-macro-support v0.2.90
...
    │               ├── wasm-bindgen-backend v0.2.90
    │               │   ├── bumpalo v3.14.0
 ...
```

1. The `wasm-bindgen-backend` package supports bumpalo `^3.0.0`
https://crates.io/crates/wasm-bindgen-backend/0.2.90/dependencies
2. The `bumpalo` package recently released some new versions that upgraded their MSRV. 
https://github.com/fitzgen/bumpalo/commit/f8597ceb3600807a902fa9692fb43c49e7b63b27

 While this is not strictly connected to this issue, I have gone ahead and bumped the MSRV to get things unblocked.

**Comment by ploftness-tesla** [comment]
@Keats any updates?  Would appreciate getting this released.

**Comment by Keats** [comment]
Thanks and sorry for the delay

**Comment by ploftness-tesla** [comment]
@Keats Thanks!

**Pull request merged**